lib/c-api/src/wasm_c_api/wasi/mod.rs

```diff
@@ -174,21 +174,28 @@ pub extern "C" fn wasi_env_new(mut config: Box<wasi_config_t>) -> Option<Box<was
 #[no_mangle]
 pub extern "C" fn wasi_env_delete(_state: Option<Box<wasi_env_t>>) {}
 
+// Dead code: deprecate or remove
 #[no_mangle]
-pub extern "C" fn wasi_env_set_instance(env: &mut wasi_env_t, instance: &wasm_instance_t) -> bool {
+pub extern "C" fn wasi_env_set_instance(
+    _env: &mut wasi_env_t,
+    _instance: &wasm_instance_t,
+) -> bool {
+    /*
     let memory = if let Ok(memory) = instance.inner.exports.get_memory("memory") {
         memory
     } else {
         return false;
     };
     env.inner.set_memory(memory.clone());
+    */
 
     true
 }
 
+// Dead code: deprecate or remove
 #[no_mangle]
-pub extern "C" fn wasi_env_set_memory(env: &mut wasi_env_t, memory: &wasm_memory_t) {
-    env.inner.set_memory(memory.inner.clone());
+pub extern "C" fn wasi_env_set_memory(_env: &mut wasi_env_t, _memory: &wasm_memory_t) {
+    //env.inner.set_memory(memory.inner.clone());
```

Comment:
Comment why this is no longer needed

Reply:
Yeah, I was putting this off for wanting to find a deprecation process but I'll add a user visible comment about it being deprecated for now and we can figure out what to do later